### PR TITLE
Improvements to opening /proc/kcore via sudo

### PR DESCRIPTION
**Pull request by osandov**
A couple of followups to #347:

- Check whether sudo is installed in the first place.
- Make the sudo prompt include why we're asking for sudo.

### CODE REVIEW

**Comment by osandov** [comment]
@brenns10 mind taking a look at a couple of usability suggestions for the sudo opener?

File: drgn/internal/sudohelper.py [hunk 1]
```diff
@@ -27,6 +27,8 @@ def open_via_sudo(
             subprocess.check_call(
                 [
                     "sudo",
+                    "-p",
+                    f"[sudo] password for %p to open {path}: ",
```
Comment (brenns10) [inline drgn/internal/sudohelper.py:31]: Today I learned about `sudo -p` :)  Turns out it's present even in OL7/RHEL7 so it's been around for a while and should work on most distros.

**Review by brenns10 — APPROVED** [review]
Looks good to me!

File: drgn/internal/sudohelper.py (hunk 1 above)
Comment (osandov) [reply drgn/internal/sudohelper.py:31]: I hadn't even considered that, but it looks like it goes way back to sudo-project/sudo@337897e2d83489b4427a90e3dba20fad28d3fe88 in 1995! So I think we're covered ;)

**Pull request merged**